server/http/protolet/protolet.go

```diff
@@ -179,3 +244,51 @@ func GenerateHTTPHandlers(servicePrefix string, server interface{}) (*HTTPHandle
 		RequestHandler:       requestHandler,
 	}, nil
 }
+
+type wrappedResponse struct {
+	w           http.ResponseWriter
+	wroteHeader bool
+	wroteBody   bool
+}
+
+func (w *wrappedResponse) Header() http.Header {
+	return w.w.Header()
+}
+
+func (w *wrappedResponse) Write(b []byte) (int, error) {
+	w.wroteBody, w.wroteHeader = true, true
+	return w.w.Write(b)
+}
+
+func (w *wrappedResponse) WriteHeader(code int) {
+	w.wroteHeader = true
+	w.w.WriteHeader(code)
+}
+
+func (w *wrappedResponse) Flush() {
+	if !w.wroteHeader && !w.wroteBody {
+		return
+	}
+	if f, ok := w.w.(http.Flusher); ok {
+		f.Flush()
+	}
+}
+
+func (w *wrappedResponse) sendErrorIfNeeded(req *http.Request) {
+	if w.wroteHeader || w.wroteBody {
+		return
+	}
+	i, err := strconv.Atoi(w.Header().Get("grpc-status"))
+	if err != nil {
+		i = int(codes.Unknown)
+	}
+	if i == 0 {
+		w.WriteHeader(200)
+	}
+
+	// Match our current behavior where we return 500 for all errors and return the message in the response body
+	w.WriteHeader(500)
```

Comment:
should there be a return after line 286?

Reply:
Good call, I sent https://github.com/buildbuddy-io/buildbuddy/pull/5509

Not sure if this ever gets hit in practice, but there's definitely supposed to be a return there.